gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -578,11 +584,20 @@ local function SelectForcedRoles(plys, selectableRoles)
 
 			selectedPlys[ply] = true
 		end
+
+		-- assigning the amount of forced players per role
+		selectedForcedRoles[subrole] = curCount
+
+		-- now assign amount of forced players per baserole if this is only a subrole
+		local baserole = roles.GetByIndex(subrole).baserole
+		if baserole and baserole ~= subrole then
+			selectedForcedRoles[baserole] = curCount
```

Comment:
Here you overwriting the amount of baseroles by the amount of subroles

Reply:
Oops, wonder why I didn't notice this while testing. Theoretically I should have gotten less baseroles, when forcing subroles.....